### PR TITLE
OM initialization fix [2/2]

### LINKED ISSUE #1972: [OM] Output manager initialized as global all across the codebase(2/2)
Same as #1848 
## Steps to Reproduce

1.
2.
3.
4.

### PR DESCRIPTION
**Pull request by matthew7838**
This PR removes the global initialization of OM in all files. Animal-related files are neglected. Some tests still uses global initialization but that should be resolved in future issues due to the complexity of those tests and the urgency of this PR. 
### Context

Issue(s) closed by this pull request: closes #1972

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?
- [ ] Yes
- [x] No

### What

- Removes global OM initialization.
- Initiates OM in static methods or constructors
- Update the tests' mockings.

### Why

- We want to avoid global variables and performance won't be much affected since OM is singleton.

### How

- Removes the global OM.

### Test plan

- Updates and runs test suites on modified code.
- Run with default data without any extra side effects.

### CODE REVIEW

**Comment by ew3361zh** [comment]
I think this looks good, Matthew but I'm still seeing some global OM initializations:
- animal_module_reporter.py
- animal_manager.py
- pen.py
- purchased_feed_emissions_estimator.py
- calf.py
- manure_manager_config_handler.py

And a number of others. I'm not sure if these were intentionally left out or still need to be addressed.

**Comment by matthew7838** [comment]
> I think this looks good, Matthew but I'm still seeing some global OM initializations:
> 
> * animal_module_reporter.py
> * animal_manager.py
> * pen.py
> * purchased_feed_emissions_estimator.py
> * calf.py
> * manure_manager_config_handler.py
> 
> And a number of others. I'm not sure if these were intentionally left out or still need to be addressed.

Yes, we will ignore the animal module since there is a major rework coming.

**Comment by ew3361zh** [comment]
> > I think this looks good, Matthew but I'm still seeing some global OM initializations:
> > 
> > * animal_module_reporter.py
> > * animal_manager.py
> > * pen.py
> > * purchased_feed_emissions_estimator.py
> > * calf.py
> > * manure_manager_config_handler.py
> > 
> > And a number of others. I'm not sure if these were intentionally left out or still need to be addressed.
> 
> Yes, we will ignore the animal module since there is a major rework coming.

sounds good. could you note that in the PR description?

Here are the non-animal module global OM intializations I'm still seeing:
- field.py
- manure_manager_config_handler.py
- a bunch in test files but I'm guessing we're not worrying about those?

File: RUFAS/routines/field/field/field.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from RUFAS.routines.manure.manure_nutrients.nutrient_request import NutrientRequest
 from RUFAS.time import Time
 
-om = OutputManager()
+# om = OutputManager()
```
Comment (ew3361zh) [inline RUFAS/routines/field/field/field.py]: Delete.

**Review by ew3361zh — APPROVED** [review]
LGTM!

**Review by allisterakun — APPROVED** [review]
Straight-forward changes, LGTM!

**Pull request merged**